integration/integration.go

```diff
@@ -125,6 +130,13 @@ func (i *Integration) AddEntity(e *Entity) {
 func (i *Integration) Publish() error {
 	defer i.Clear()
 
+	hostID := i.GetHostID()
+	if hostID != "" {
+		for k := range i.Entities {
+			i.Entities[k].Metadata.Name = replaceLocalhost(i.Entities[k].Metadata.Name, hostID)
```

Comment:
Do all entities always have metadata? If not we should check if it's not nil before accessing it. Maybe is not a common/real scenario but just in case.

Reply:
All excepting the localhost one if they are created using the constructor https://github.com/newrelic/infra-integrations-sdk/blob/786cf26d3cdd3b8701c686be3f6a7ea25bc784b2/integration/integration.go#L105 

But as it is an exported struct I would also check it.

Reply:
added check from Metadata